warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reindex/ShardReindexMapper.java

```diff
@@ -0,0 +1,764 @@
+package datawave.ingest.mapreduce.job.reindex;
+
+import static datawave.ingest.data.config.DataTypeHelper.Properties.DATA_NAME;
+import static org.apache.commons.lang3.StringUtils.reverse;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.hadoop.conf.Configurable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.WritableComparator;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.log4j.Logger;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
+import datawave.data.hash.HashUID;
+import datawave.ingest.config.RawRecordContainerImpl;
+import datawave.ingest.data.RawRecordContainer;
+import datawave.ingest.data.Type;
+import datawave.ingest.data.TypeRegistry;
+import datawave.ingest.data.config.NormalizedContentInterface;
+import datawave.ingest.data.config.ingest.IngestHelperInterface;
+import datawave.ingest.mapreduce.ContextWrappedStatusReporter;
+import datawave.ingest.mapreduce.handler.DataTypeHandler;
+import datawave.ingest.mapreduce.handler.shard.ShardedDataTypeHandler;
+import datawave.ingest.mapreduce.job.BulkIngestKey;
+import datawave.ingest.mapreduce.partition.MultiTableRangePartitioner;
+import datawave.ingest.protobuf.Uid;
+
+public class ShardReindexMapper extends Mapper<Key,Value,BulkIngestKey,Value> {
+    public static final String CLEANUP_SHARD = "cleanupShard";
```

Comment:
Thoughts on prefixing these? It can be nice to search the GUI for `ShardReindexMapper` and see all the applicable options.

Reply:
Sure. The beauty of constant is it doesn't matter